### ISSUE TITLE
Detect unnecessary empty comment lines in YARD documentation blocks

### ISSUE THREAD

**Issue by nijikon** [mensfeld/yard-lint#28]
## Summary

Add a new check to detect and report empty comment lines (`#` or `# `) at the beginning and end of YARD documentation blocks.

## Current Behavior

yard-lint currently does not flag empty comment lines that pad YARD documentation blocks. The following passes without warnings:

```ruby
#
# @param organization_id [String]
# @param id [String]
#
# @return [Pet]
#
def call(organization_id, id)
```

## Proposed Behavior

yard-lint should detect and report:

1. **Leading empty comment lines** - Empty `#` lines at the start of a documentation block (before any `@tag` or description text)
2. **Trailing empty comment lines** - Empty `#` lines immediately before the method/class definition

The above example would trigger warnings, and the expected clean format would be:

```ruby
# @param organization_id [String]
# @param id [String]
#
# @return [Pet]
def call(organization_id, id)
```

## Rationale

- **Consistency**: Enforces a uniform documentation style across a codebase
- **Readability**: Reduces visual noise - the empty lines add no semantic value
- **Convention alignment**: Matches common Ruby documentation style guides that recommend starting documentation immediately with content
- **Space efficiency**: Especially valuable in files with many documented methods

### New error types

- `EmptyLeadingCommentLine` - For empty comment lines at the start of a YARD block
- `EmptyTrailingCommentLine` - For empty comment lines at the end of a YARD block (before the definition)

### Configuration (optional)

Could be configurable to allow users to opt-in/out:

```yaml
# .yard-lint.yml
EmptyLeadingCommentLine:
  enabled: true
EmptyTrailingCommentLine:
  enabled: true
```

### Edge cases to consider

- Empty lines **between** tags (e.g., between `@param` and `@return` blocks) should likely remain allowed, as they improve readability by grouping related tags
- Documentation blocks with only a description (no tags) that intentionally use empty lines for paragraph separation

### Should warn (leading empty line)

```ruby
#
# @param id [String]
def find(id)
```

### Should warn (trailing empty line)

```ruby
# @param id [String]
#
def find(id)
```

### Should NOT warn (empty line between tag groups is acceptable)

```ruby
# @param organization_id [String]
# @param id [String]
#
# @return [Pet]
def call(organization_id, id)
```

### Should NOT warn (clean format)

```ruby
# @param id [String]
#
# @return [Pet]
def find(id)
```

## Related

This is a stylistic enhancement similar to how RuboCop's `Layout/EmptyLinesAroundMethodBody` works for method bodies, but applied to YARD documentation blocks.
